lib/ansible/cli/doc.py

```diff
@@ -793,15 +793,20 @@ def namespace_from_plugin_filepath(filepath, plugin_name, basedir):
     @staticmethod
     def _get_plugin_doc(plugin, plugin_type, loader, search_paths):
         # if the plugin lives in a non-python file (eg, win_X.ps1), require the corresponding python file for docs
-        result = loader.find_plugin_with_context(plugin, mod_type='.py', ignore_deprecated=True, check_aliases=True)
-        if not result.resolved:
-            raise PluginNotFound('%s was not found in %s' % (plugin, search_paths))
+        for ext in ('.py', '.yml', '.yaml'):
```

Comment:
For anything new, would strongly suggest we pick `.yml` and only `.yml`. Doubling up the file extensions causes all sorts of interesting precedence problems, especially the way this is being done.

Reply:
well, precedence is easy to declare, the problem is many people will want to choose the extension, in the end this will either be a constant and/or configurable setting (leaning to no)